src/main/java/org/gradle/profiler/GradleBuildInvocationResult.java

```diff
@@ -55,32 +57,39 @@ public String getName() {
             }
 
             @Override
-            public Duration extractFrom(GradleBuildInvocationResult result) {
-                return result.cumulativeBuildOperationTimes.getOrDefault(buildOperationDetailsClass, Duration.ZERO);
+            public Duration extractTotalDurationFrom(GradleBuildInvocationResult result) {
+                BuildOperationDuration duration = result.cumulativeBuildOperationDurations.get(buildOperationDetailsClass);
+                return duration == null ? Duration.ZERO : duration.getTotalDuration();
+            }
```

Comment:
Nit: we could define a `BuildOperationDuration.ZERO` constant, and use it like we did with `Duration.ZERO`.

Reply:
Good idea! I also realized that `BuildOperationDuration` can be immutable too, given the usages

Reply:
Done with a slight refactoring to make `BuildOperationExecutionData` immutable and accompanied by a builder.